akka-sample-persistence-dc-scala/src/main/scala/sample/persistence/res/counter/ThumbsUpCounter.scala

```diff
@@ -1,40 +1,34 @@
-package sample.persistence.multidc
+package sample.persistence.res.counter
 
 import akka.actor.typed.scaladsl.Behaviors
 import akka.actor.typed.{ActorRef, Behavior}
 import akka.cluster.sharding.typed.ReplicatedEntityProvider
 import akka.persistence.cassandra.query.scaladsl.CassandraReadJournal
-import akka.persistence.typed.scaladsl.{Effect, EventSourcedBehavior, ReplicatedEventSourcing, ReplicationContext}
+import akka.persistence.typed.scaladsl.{Effect, EventSourcedBehavior, ReplicatedEventSourcing}
 import akka.persistence.typed.{ReplicaId, ReplicationId}
+import sample.persistence.res.{CborSerializable, MainApp}
 
 object ThumbsUpCounter {
 
   // sent over sharding
-  sealed trait Command extends CborSerializer {
+  sealed trait Command extends CborSerializable {
     def resourceId: String
   }
 
   final case class GiveThumbsUp(resourceId: String, userId: String, replyTo: ActorRef[Long]) extends Command
-
   final case class GetCount(resourceId: String, replyTo: ActorRef[Long]) extends Command
-
   final case class GetUsers(resourceId: String, replyTo: ActorRef[State]) extends Command
 
   // saved to DB
-  sealed trait Event extends CborSerializer
-
+  sealed trait Event extends CborSerializable
   final case class GaveThumbsUp(userId: String) extends Event
 
   // saved to DB
-  final case class State(users: Set[String]) extends CborSerializer {
+  final case class State(users: Set[String]) extends CborSerializable {
     def add(userId: String): State = copy(users + userId)
   }
 
-  // In this sample one replica per DC
-  val Replicas = Set(ReplicaId("eu-west"), ReplicaId("eu-central"))
-
-  val Provider: ReplicatedEntityProvider[Command] = ReplicatedEntityProvider.perDataCenter("counter", Replicas) { replicationId => ThumbsUpCounter(replicationId) }
-
+  val Provider: ReplicatedEntityProvider[Command] = ReplicatedEntityProvider.perDataCenter("counter", MainApp.AllReplicas) { replicationId => ThumbsUpCounter(replicationId) }
```

Comment:
On the other hand that adds a bit of complexity if all you are after is to understand replicated entities. Maybe good to keep it focused on the insides of replicated entities?

(but perhaps worth noting that in the other examples then "this would normally be used with sharding, see thumbs up for an example of that")

Reply:
i'll add the providers as they are one liners now but won't add any HTTP endpoints